core/state/rw_v3.go

```diff
@@ -194,7 +194,7 @@ func (rs *StateV3) ApplyState4(ctx context.Context, txTask *TxTask) error {
 		return fmt.Errorf("StateV3.ApplyLogsAndTraces: %w", err)
 	}
 
-	if (txTask.TxNum+1)%rs.domains.StepSize() == 0 /*&& txTask.TxNum > 0 */ {
+	if (txTask.TxNum+1)%rs.domains.StepSize() == 0 || (txTask.BlockNum)%32 == 0 /*&& txTask.TxNum > 0 */ {
```

Comment:
1. i'm not sure it's safe to merge
2. probably need change to `(txTask.Final && txTask.BlockNum%32 == 0)`

Reply:
for sure not safe, thanks for pointing on final